snap/hooks/install

```diff
@@ -14,8 +14,8 @@ mkdir -p $DATA
 
 cp "${SNAP}"/etc/$COMPONENT/* $CONF
 
-chmod -R 770 "${SNAP_DATA}"/*
-chmod -R 770 "${SNAP_COMMON}"/*
+chmod -R 775 "${SNAP_DATA}"/*
+chmod -R 775 "${SNAP_COMMON}"/*
```

Comment:
**question:** Why? Convenience so we don't have to run in sudo?

If so, I don't think we should do this change.

Reply:
TLDR: This is indeed what has been bugging me a bit and it is actually one of the main reason for the PR being WIP, as I was not quite happy/sure with this yet.

The reason is that I believe we should probably provide a bit better (and more granular) permission settings:

* $SNAP_COMMON/var/lib/ => This is very sensitive data, which should be 700, as it really only needs to be read/write by the `daemon` user. 
* $SNAP_COMMON/var/log/ => This is not very sensitive data, that should be written only by `daemon` but should be read ideally by anyone in the `adm` group (although I suppose in the context of snap the `snap_daemon` group is used instead), therefore 750 
* $SNAP_COMMON/etc/kafka => This **should not* be very sensitive data, and that - in general - it should be fine to allow general people to read and people in the group to write, therefore 775. I'm saying *should not* because one file that is a bit borderline is `server.properties`, since it can contain passwords unfortunately. Also, here we generally store trustore/keystore/private keys/etc, for which we should have even stricter permissions. I believe here it is fine to have permission at 775, but with the charms handling permission of single files a bit more restrictively. 

Also the reason why I believe it would be appropriate to have 775 for $SNAP_COMMON/etc/kafka is that here there exist also files that you want - as external user - to be able to read. In the snap we have two different entrypoints:
* charmed-kafka.daemon => that runs as `snap_daemon`, so no problem to read those files
* charmed-kafka.<commands> => that runs as the user
I believe, for a smooth UX, any user should be able to run the `charmed-kafka.<commands>` without being `sudo`.